### PR TITLE
feat(elements): Add `validateRegisteredStrategies` action to verification machine

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

Add verifications warning/error messages when building a verifications flow.

When `supportedFirstFactors` does not match `registeredStrategies` add a warning about the inconsistency

```
Clerk: Your instance is configured to support strategies: password, email_code, email_link, reset_password_email_code, but the rendered strategies are: passwords, email_link, email_code, reset_password_email_code. Before deploying your app, make sure to render a <Strategy> component for each supported strategy. For more information, visit the documentation: <link>
```

Screenshot 2024-04-10 at 12 35 01 PM

When trying to render the current factor, if the user has not added the corresponding `<Strategy name=".." />` component, throw a runtime error

```
ClerkElementsRuntimeError: Your sign-in/up attempt is missing a password strategy. Make sure <Strategy name="password"> is rendered in your flow. For more information, visit the documentation: <link>
```

Screenshot 2024-04-10 at 12 35 20 PM

A follow up task will add similar validation to the steps component rendering.

https://linear.app/clerk/issue/SDK-1426/[dx]-improve-error-warning-messaging-when-building-a-flow

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts [hunk 1]
```diff
@@ -53,6 +54,38 @@ const SignInVerificationMachine = setup({
...
+        )
+      ) {
+        console.warn(
+          `Clerk: Your instance is configured to support strategies: ${clerk.client.signIn.supportedFirstFactors
+            .map(f => f.strategy)
+            .join(', ')}, but the rendered strategies are: ${[...context.registeredStrategies]
+            .map(s => s)
+            .join(
+              ', ',
+            )}. Before deploying your app, make sure to render a <Strategy> component for each supported strategy. For more information, visit the documentation: <link>`,
```
Comment (alexcarpenter) [inline packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts:76]: TODO: Add link to docs

**Review by tmilewski — APPROVED** [review]

File: packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts (hunk 1 above)
Comment (BRKalow) [reply packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts:76]: Most likely: `https://clerk.com/docs/elements/reference/sign-in#strategy`

File: packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts [hunk 2]
```diff
@@ -53,6 +54,38 @@ const SignInVerificationMachine = setup({
...
+        );
+      }
+
+      if (
+        process.env.NODE_ENV === 'development' &&
+        !context.registeredStrategies.has(context.currentFactor?.strategy as unknown as SignInFactor)
+      ) {
+        throw new ClerkElementsRuntimeError(
+          `Your sign-in/up attempt is missing a ${context.currentFactor?.strategy} strategy. Make sure <Strategy name="${context.currentFactor?.strategy}"> is rendered in your flow. For more information, visit the documentation: <link>`,
```
Comment (BRKalow) [inline packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts:85]: This machine is only for sign-in, so we can get rid of `/up`.

Comment (alexcarpenter) [reply packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts]: https://github.com/clerk/javascript/pull/3165/commits/543291c6266368749103d410c40e038f30bb7017

File: packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts (hunk 1 above)
Comment (alexcarpenter) [reply packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts]: https://github.com/clerk/javascript/pull/3165/commits/ef491eefa328185725d77ef84b3f968f5519fa33

File: packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts [hunk 3]
```diff
@@ -53,6 +54,38 @@ const SignInVerificationMachine = setup({
...
     },
+    validateConfiguredStrategies: ({ context }) => {
+      const clerk = context.parent.getSnapshot().context.clerk;
+
+      if (clerk.__unstable__environment?.isProduction()) {
+        return;
+      }
+
+      if (
+        !clerk.client.signIn.supportedFirstFactors.every(factor =>
```
[the remark this answers is not in the document]
Comment (BRKalow) [reply packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts:65]: Yep, I think so. We don't distinguish between the two in the component API.

[the remark this answers is not in the document]
Comment (alexcarpenter) [reply packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts:65]: https://github.com/clerk/javascript/pull/3165/commits/b38d3a9def1881d4b4eb172ece35de822acf7e93

File: packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts [hunk 4]
```diff
@@ -53,6 +54,57 @@ const SignInVerificationMachine = setup({
...
+          context.registeredStrategies.has(factor.strategy as unknown as SignInFactor),
+        )
+      ) {
+        console.warn(
+          `Clerk: Your instance is configured to support strategies: ${clerk.client.signIn.supportedFirstFactors
```
Comment (BRKalow) [inline packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts:70]:
```suggestion
          `Clerk: Your instance is configured to support these strategies: ${clerk.client.signIn.supportedFirstFactors
```

**Review by BRKalow — APPROVED** [review]

**Pull request merged**